lib/Onnxifi/GlowOnnxifiManager.cpp

```diff
@@ -93,21 +93,23 @@ GlowOnnxifiManager::createGraph(BackendPtr backend,
   return graph;
 }
 
-runtime::HostManager *
+std::shared_ptr<runtime::HostManager>
 GlowOnnxifiManager::getOrCreateHostManager(BackendKind backendKind) {
+  std::shared_ptr<runtime::HostManager> hostManager;
+
   auto it = hostManagers_.find(backendKind);
 
-  // If a HostManager doesn't exist yet for this BackendKind then create one
-  if (it == hostManagers_.end()) {
-    auto newHostManager =
-        onnxifi::HostManagerBackendId::createHostManager(backendKind);
-    assert(newHostManager != nullptr);
-    auto ptr = newHostManager.get();
-    hostManagers_[backendKind] = std::move(newHostManager);
-    return ptr;
+  if (it != hostManagers_.end()) {
+    hostManager = it->second.lock();
```

Comment:
If this is null (because the object has been destroyed) we should probably remove it from the `hostManagers_` map.

Reply:
There's not much point, if it's null it will be replaced in the map on line 109